app/forms/batch_upsert_holidays.rb

```diff
@@ -0,0 +1,63 @@
+class BatchUpsertHolidays
+  include ActiveModel::Model
+  extend ActiveModel::Naming
+  include DateRangePickerHelper
+
+  attr_reader :title
+  attr_reader :date_range
+  attr_reader :users
+  attr_reader :all_day
+
+  validates :title, presence: true
+  validates :date_range, presence: true
+  validates :users, presence: true
+
+  def to_model
+    if @previous_holiday_ids.any?
+      @previous_holiday ||= Holiday.find(@previous_holiday_ids.first)
```

Comment:
Is it me or is it a bit surprising that you're memoizing based on a variable with a different name to the accessor?